internal/appsec/trace/tagsholder.go

```diff
@@ -5,29 +5,64 @@
 
 package trace
 
-import "sync"
+import (
+	"encoding/json"
+	"sync"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+type serializableTag struct {
+	tag any
+}
+
+func (t *serializableTag) MarshalJSON() ([]byte, error) {
```

Comment:
The struct contains a pointer and is also small enough to pass it by copy to avoid an extra alloc:

```suggestion
func (t serializableTag) MarshalJSON() ([]byte, error) {
```